src/components/DropdownProblem.tsx

```diff
@@ -1,7 +1,91 @@
 import { BaseProblemProps } from './ProblemSetBlock'
+import { Formik, Form, Field, ErrorMessage } from 'formik'
+import { mathifyElement } from '../lib/math'
+import React, { useCallback, useState } from 'react'
+import * as Yup from 'yup'
 
-interface DropdownProblemProps extends BaseProblemProps {}
+interface DropdownProblemProps extends BaseProblemProps {
+  solutionOptions: string
+}
+
+interface DropdownFormValues {
+  response: string
+}
+
+export const DropdownProblem = ({
+  solvedCallback, exhaustedCallback, allowedRetryCallback, content, buttonText, solutionOptions,
+  encourageResponse, correctResponse, solution, retryLimit
+}: DropdownProblemProps): JSX.Element => {
+  const [feedback, setFeedback] = useState('')
+  const [formDisabled, setFormDisabled] = useState(false)
+  const [retriesAllowed, setRetriesAllowed] = useState(0)
+
+  const schema = Yup.object({
+    response: Yup.string().trim().required('Please select an answer')
+  })
+
+  const generateOptions = (): JSX.Element[] => {
+    const options: JSX.Element[] = []
+    const parsedOptionValues: string[] = JSON.parse(solutionOptions)
+
+    options.push(<option key="initial" value="">Select an answer</option>)
+    parsedOptionValues.forEach(val => options.push(<option key={val} value={val}>{val}</option>))
+
+    return options
+  }
+
+  const contentRefCallback = useCallback((node: HTMLDivElement | null): void => {
+    if (node != null) {
+      mathifyElement(node)
+    }
+  }, [])
+
+  const clearFeedback = (): void => {
+    setFeedback('')
+  }
+
+  const handleSubmit = async (values: DropdownFormValues): Promise<void> => {
+    if (values.response === solution) {
+      setFeedback(correctResponse)
+      solvedCallback()
+      setFormDisabled(true)
+    } else if (retryLimit === 0 || retriesAllowed !== retryLimit) {
+      setRetriesAllowed(currRetries => currRetries + 1)
+      setFeedback(encourageResponse)
+      allowedRetryCallback()
+    } else {
+      exhaustedCallback()
+      setFeedback('No more attempts allowed')
+      setFormDisabled(true)
+    }
+  }
 
-export const DropdownProblem = ({ solvedCallback, exhaustedCallback, allowedRetryCallback }: DropdownProblemProps): JSX.Element => {
-  return <></>
+  return (
+    <div className="os-raise-bootstrap">
+      <div className="my-3" ref={contentRefCallback} dangerouslySetInnerHTML={{ __html: content }} />
```

Comment:
What is the significance of `my-3` and later `os-form-select mb-3`

Reply:
These are Bootstrap classes for styling (I'm trying to minimize any styling on this iteration, but even I have my tolerance threshold where I'm thinking "that's just too ugly" :joy: ). 

The `my-3` and `mb-3` are utility classes that set margins per [here](https://getbootstrap.com/docs/5.1/utilities/spacing/). `my-3` set top and bottom margins and `mb-3` just bottom. Basically this is so things aren't all scrunched together.

The `os-form-select` class is a slight change to the `form-select` class described [here](https://getbootstrap.com/docs/5.1/forms/select/). I wanted to add some styling as the default unstyled dropdown is hideous:

![image](https://user-images.githubusercontent.com/839025/168094990-c8d63a28-669b-44ab-86f4-5dadfadb1010.png)

At the same time the bootstrap`form-select` has `width: 100%` per [here](https://github.com/twbs/bootstrap/blob/88a6610895c05d05f4710d0783eb5ea2f30294fe/scss/forms/_form-select.scss#L8) which looks like:

![image](https://user-images.githubusercontent.com/839025/168095238-05bc0f26-6795-4c10-ae3f-00f460487eca.png)

That's why this PR includes a custom `.os-form-select` class which extends the `.form-select` but unsets `width` so it displays more like:

![image](https://user-images.githubusercontent.com/839025/168095679-f6bfd2e7-e5df-4c18-8598-2baa310e4bd6.png)


Again, these are quick cosmetic changes so things aren't too grotesque. I fully expect a lot of this will get revisited once Maddy has a chance to stare at examples with real content and give some UX feedback.